packages/@sanity/cli/src/util/journeyConfig.ts

```diff
@@ -0,0 +1,270 @@
+  validateProjectId(projectId)
+  try {
+    // TODO: Add token to the request and change to sanity.io
+    const response = await import(
+      `https://api.sanity.work/v2024-02-23/journey/projects/${projectId}/schema`
+    )
+
+    return response.default
+  } catch (err) {
+    console.error(err)
+    throw new Error(`Failed to fetch remote schema: ${projectId}`)
+  }
+}
+
+/**
+ * Wrap a Journey schema in a module export
+ *
+ * @param schema - The Journey schema to wrap in a module export
+ * @returns The Journey schema as a module export
+ */
+async function JourneySchemaToFileContents(schemaType: DocumentOrObject): Promise<string> {
+  const serialised = wrapSchemaTypeInHelpers(schemaType)
+  const imports = getImports(serialised)
+  const prettifiedSchemaType = await format(serialised, {parser: 'typescript'})
+  // Start file with import, then export the schema type as a named export
+  return `${imports}\n\nexport const ${schemaType.name} = ${prettifiedSchemaType}\n`
+}
+
+/**
+ * Assemble a list of Journey schema module exports into a single index file
+ *
+ * @param schemas - The Journey schemas to assemble into an index file
+ * @returns The index file as a string
+ */
+function assembeJourneyIndexContent(schemas: DocumentOrObject[]): string {
+  schemas.sort((a, b) => (a.name > b.name ? 1 : -1)) // Sort schemas alphabetically by name
+  const imports = schemas.map((schema) => `import { ${schema.name} } from './${schema.name}'`)
+  const exports = schemas.map((schema) => `  ${schema.name}`).join(',\n')
+  return `${imports.join('\n')}\n\nexport const schemaTypes = [\n${exports}\n]`
+}
+
+/**
+ * Get the import statements for a schema type
+ *
+ * @param schemaType - The schema type to get the imports for
+ * @returns The import statements for the schema type
+ */
+function getImports(schemaType: string): string {
+  const defaultImports = ['defineType', 'defineField']
+  if (schemaType.includes('defineArrayMember')) {
+    defaultImports.push('defineArrayMember')
+  }
+  return `import { ${defaultImports.join(', ')} } from 'sanity'`
+}
+
+/**
+ * Serialize a Sanity schema type into a string.
+ * Wraps the schema object in the appropriate helper function.
+ *
+ * @param schemaType - The schema type to serialize
+ * @returns The schema type as a string
+ */
+function wrapSchemaTypeInHelpers(schemaType: SchemaObject): string {
+  switch (schemaType.type) {
+    case 'document':
+    case 'object':
+      return wrapDefineType(schemaType)
+    case 'array':
+      return wrapDefineArrayMember(schemaType as ArrayDefinition)
+    default:
+      return wrapDefineField(schemaType)
+  }
+
+  function wrapDefineType(field: SchemaObject) {
+    const {fields, ...rest} = field
+    const restPart = serialize(rest)
+    const fieldsStr = fields?.map(wrapSchemaTypeInHelpers).join('')
+    const fieldsPart = fields ? `fields: [${fieldsStr}],` : ''
+    return `defineType({
```

Comment:
Is this needed now that the schemas are generated using `defineType`?

Reply:
I wish we didn't. We still need to serialise the schema types to be able to save them to a file as we can't save a JS object. This means re-adding all of the helpers once we have consumed the ESM

Reply:
ah, of course, we're consuming the output here.